src/lib/browse/LayerControls.svelte

```diff
@@ -72,6 +73,28 @@
       return getRoadLayerNames(map, get(mapStyle));
     },
   };
+
+  const initialQueryParams = new URLSearchParams(window.location.search);
+
+  if (
+    !initialQueryParams ||
+    (initialQueryParams.size === 1 &&
+      initialQueryParams.keys().next().value == "style")
+  ) {
+    let url = new URL(window.location.href);
+    const previousSessionQueryString =
+      window.localStorage.getItem(localStorageQuerystringKey) || "";
+    const newQueryParams = new URLSearchParams(previousSessionQueryString);
+    // @ts-ignore typescript is convinced foreach doesn't exist but it's working?
+    newQueryParams.entries().forEach(([key, value]) => {
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/forEach works fine, no `entries`. Or do regular `for` loops to remain consistent with the rest of the codebase